sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapp/AadOidcIdTokenDecoderFactory.java

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.aad.implementation.webapp;
+
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.core.oidc.OidcIdToken;
+import org.springframework.security.oauth2.jose.jws.SignatureAlgorithm;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.security.oauth2.jwt.JwtDecoderFactory;
+import org.springframework.security.oauth2.jwt.NimbusJwtDecoder;
+import org.springframework.web.client.RestOperations;
+
+/**
+ * A factory that provides a {@link JwtDecoder} used for {@link OidcIdToken} signature verification.
+ *
+ * @see <a href="https://learn.microsoft.com/azure/active-directory/develop/id-tokens">azure-active-directory id-tokens</a>
+ */
+public class AadOidcIdTokenDecoderFactory implements JwtDecoderFactory<ClientRegistration> {
```

Comment:
Shall we try to delegate the jwt decoder factory to `org.springframework.security.oauth2.client.oidc.authentication.OidcIdTokenDecoderFactory`?

Reply:
Already tried but didn't find a good way to achieve this. Already tried these ways:

1. Use extends: **OidcIdTokenDecoderFactory** is a final class, so it cannot be extended.
2. Use delegate:
    2.1. The returned value of **OidcIdTokenDecoderFactory** is **JwtDecoder**, not sure to be a **NimbusJwtDecoder** (It's possible that **OidcIdTokenDecoderFactory** return other type of **JwtDecoder**).
    2.2. Even we assume that the returned value of **OidcIdTokenDecoderFactory** is **NimbusJwtDecoder** and force convert to **NimbusJwtDecoder**, the **NimbusJwtDecoder** cannot set **RestTemplate**.